app/presentation/ImagePinch.js

```diff
@@ -0,0 +1,440 @@
+		state.position
+	];
+}
+
+function runDecay(clock, value, velocity) {
+	const state = {
+		finished: new Value(0),
+		velocity: new Value(0),
+		position: new Value(0),
+		time: new Value(0)
+	};
+
+	const config = { deceleration: 0.99 };
+
+	return [
+		cond(clockRunning(clock), 0, [
+			set(state.finished, 0),
+			set(state.velocity, velocity),
+			set(state.position, value),
+			set(state.time, 0),
+			startClock(clock)
+		]),
+		set(state.position, value),
+		decay(clock, state, config),
+		cond(state.finished, stopClock(clock)),
+		state.position
+	];
+}
+
+function bouncyPinch(
+	value,
+	gesture,
+	gestureActive,
+	focalX,
+	displacementX,
+	focalY,
+	displacementY
+) {
+	const clock = new Clock();
+
+	const delta = scaleDiff(gesture);
+	const rest = scaleRest(value);
+	const focalXRest = cond(
+		lessThan(value, 1),
+		0,
+		sub(displacementX, multiply(focalX, add(-1, divide(rest, value))))
+	);
+	const focalYRest = cond(
+		lessThan(value, 1),
+		0,
+		sub(displacementY, multiply(focalY, add(-1, divide(rest, value))))
+	);
+	const nextScale = new Value(1);
+
+	return cond(
+		[delta, gestureActive],
+		[
+			stopClock(clock),
+			set(nextScale, scaleFriction(value, rest, delta)),
+			set(
+				displacementX,
+				sub(displacementX, multiply(focalX, add(-1, divide(nextScale, value))))
+			),
+			set(
+				displacementY,
+				sub(displacementY, multiply(focalY, add(-1, divide(nextScale, value))))
+			),
+			nextScale
+		],
+		cond(
+			or(clockRunning(clock), not(eq(rest, value))),
+			[
+				set(displacementX, runTiming(clock, displacementX, focalXRest, false)),
+				set(displacementY, runTiming(clock, displacementY, focalYRest, false)),
+				runTiming(clock, value, rest)
+			],
+			value
+		)
+	);
+}
+
+function bouncy(
+	value,
+	gestureDiv,
+	gestureActive,
+	lowerBound,
+	upperBound,
+	f
+) {
+	const timingClock = new Clock();
+	const decayClock = new Clock();
+
+	const velocity = speed(value);
+
+	// did value go beyond the limits (lower, upper)
+	const isOutOfBounds = or(
+		lessThan(value, lowerBound),
+		lessThan(upperBound, value)
+	);
+	// position to snap to (upper or lower is beyond or the current value elsewhere)
+	const rest = cond(
+		lessThan(value, lowerBound),
+		lowerBound,
+		cond(lessThan(upperBound, value), upperBound, value)
+	);
+	// how much the value exceeds the bounds, this is used to calculate friction
+	const outOfBounds = abs(sub(rest, value));
+
+	return cond(
+		[gestureDiv, velocity, gestureActive],
+		[
+			stopClock(timingClock),
+			stopClock(decayClock),
+			add(value, divide(gestureDiv, f(outOfBounds)))
+		],
+		cond(
+			or(clockRunning(timingClock), isOutOfBounds),
+			[stopClock(decayClock), runTiming(timingClock, value, rest)],
+			cond(
+				or(clockRunning(decayClock), lessThan(5, abs(velocity))),
+				runDecay(decayClock, value, velocity),
+				value
+			)
+		)
+	);
+}
+
+const WIDTH = 300;
+const HEIGHT = 300;
+
+class ImagePinch extends ResponsiveComponent {
```

Comment:
Can we rename it to `ImageViewer` and comment it was picked from https://github.com/software-mansion/react-native-reanimated/tree/master/Example/imageViewer?